3-UsingAPIs/Week2/assignment/ex1-programmerFun/index.js

```diff
@@ -16,29 +16,39 @@ Full description at: https://github.com/HackYourFuture/Assignments/blob/main/3-U
    url with `.shx`. There is no server at the modified url, therefore this 
    should result in a network (DNS) error.
 ------------------------------------------------------------------------------*/
-function requestData(url) {
-  // TODO return a promise using `fetch()`
+async function requestData(url) {
+
+  const response = await fetch(url);
+  
+  if (!response.ok) {
+    throw new Error("some error with API" +  response.status);
```

Comment:
Pay attention to detail: for instance, if the `status` code is 404 your message would print as "some error with API404". Also, avoid using vague, unhelpful messages like "some error".

```suggestion
    throw new Error("HTTP error: " +  response.status);
```

Note the space after the colon.